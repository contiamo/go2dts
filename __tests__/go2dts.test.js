const go2dts = require("../src/index");
const rimraf = require("rimraf");
const { join } = require("path");
const { readFileSync, readdirSync } = require("fs");

beforeAll(next => {
  rimraf(join(__dirname, "./outputs"), () => {
    go2dts(join(__dirname, "./inputs"), join(__dirname, "./outputs"));
    next();
  });
});

describe("go2dts", () => {
  readdirSync(join(__dirname, "./outputs"))
    .filter(fileName => /^[a-z]+(?!test)\.d.ts$/.test(fileName))
    .forEach(fileName => {
      describe(fileName, () => {
        it("should match the snapshot", () => {
          expect(
            readFileSync(join(__dirname, "./outputs", fileName), "utf-8")
          ).toMatchSnapshot();
        });
      });
    });
});
