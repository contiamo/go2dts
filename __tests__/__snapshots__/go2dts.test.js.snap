// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`go2dts should match the snapshot 1`] = `
"// Generated by go2dts

export type Time = string

export type UUID = string

export interface RegisterBundleRequest {
  name: string
  gitUrl: string
  branch: string
  tenantId: string
  realmId: string
}

export interface PatchBundleRequest {
  name?: string
  gitUrl?: string
  branch?: string
  coverImageURL?: string
  tags: string[]
}

export interface BundleResponse {
  id: UUID
  createdAt: Time
  updatedAt: Time
  syncedAt: Time | null
  deployedAt: Time | null
  deployedSHA: string
  name: string
  gitUrl: string
  gitSHA: string
  branch: string
  config: Bundle
  publicKey: string
  coverImageURL: string
  tags: string[]
  tenantId: string
  realmId: string
  activeDeployID: UUID | null
  topContributors: Contributor[]
  description: string
}

export interface BundleListResponse {
  page: PageMeta
  data?: BundleResponse[]
}

export interface DeployResponse {
  bundle: BundleResponse
  functions?: FunctionResponse[]
}

export interface DeployLog {
  id: UUID
  createdAt: Time
  bundleId: UUID
  deployId: UUID
  gitUrl: string
  sha: string
  message: string
  status: string
}

export interface DeployLogResponse {
  page: PageMeta
  data: DeployLog[]
}

export interface ContributorListResponse {
  page: PageMeta
  data?: Contributor[]
}

export interface BundleEditStartResponse {
  url: string
}

export interface BundleSyncResponse {
  sha: string
  branch: string
  gitURL: string
  updated: boolean
}

export interface jsonErrorMessage {
  errors: string[]
  fieldErrors: {[key: string]: string[]}
}



export type EditorStageStatus = \\"todo\\" | \\"running\\" | \\"done\\" | \\"failed\\"

export type EditorState = \\"unknown\\" | \\"failed\\" | \\"starting\\" | \\"does not exist\\" | \\"running\\" | \\"stopping\\" | \\"stopped\\" | \\"resuming\\" | \\"destroying\\"

export interface EditorStatus {
  name: string
  realmID: string
  bundleID: string
  createdAt?: Time
  lastActiveAt?: Time
  status: EditorState
  statusMessage: string
  stages: EditorStage[]
  editorURL: string
}

export interface EditorListResponse {
  page: PageMeta
  data?: EditorStatus[]
}

export interface EditorStage {
  message: string
  status: EditorStageStatus
}

export interface FunctionResponse {
  id?: UUID
  createdAt: Time
  updatedAt: Time
  bundleId: UUID
  name: string
  image: string
  command: string
  environment: {[key: string]: string}
  secrets: string[]
  schema: any
  deployed: boolean
  deploymentStatus: string
  url: string
}

export interface FunctionInstanceResponse {
  id?: UUID
  createdAt: Time
  updatedAt: Time
  bundleId: UUID
  name: string
  image: string
  command: string
  environment: {[key: string]: string}
  secrets: string[]
  schema: any
  deployed: boolean
  deploymentStatus: string
  url: string
}

export interface FunctionListResponse {
  page: PageMeta
  data?: FunctionResponse[]
}

export type TriggerType = \\"user\\" | \\"apikey\\" | \\"schedule\\" | \\"webhook\\" | \\"unknown\\"

export type ExecutionState = \\"unknown\\" | \\"failed\\" | \\"success\\" | \\"running\\"

export interface JobResponse {
  id: string
  lastRunAt?: Time
  bundleID: UUID
  url: string
  name: string
  description: string
  notebookPath: string
  schedule: string
  environment: {[key: string]: string}
  secrets: string[]
  success: boolean
  internal: boolean
}

export interface JobListResponse {
  data?: JobResponse[]
}

export interface JobRunRequest {
  environment: {[key: string]: string}
}

export interface JobRunResponse {
  executionID: string
}

export interface JobSpecification {
  schedule: string
  image: string
  command: string[]
  environment: {[key: string]: string}
  secrets: string[]
}

export interface TriggeredBy {
  type: TriggerType
  id: string
  name: string
}

export interface ExecutionResponse {
  id: string
  jobID: string
  bundleID: string
  jobName: string
  startedAt: Time
  completedAt?: Time
  success: boolean
  status: ExecutionState
  specification: JobSpecification
  triggeredBy: TriggeredBy
}

export interface ExecutionStatus {
  startedAt: Time
  completedAt?: Time
  success: boolean
}

export interface ExecutionListResponse {
  data?: ExecutionResponse[]
}

export interface LogMessage {
  timestamp: Time
  msg: string
}

export interface LogsResponse {
  status: ExecutionStatus
  data: LogMessage[]
}

export interface PageMeta {
  next?: number
  prev?: number
  last: number
  count: number
}

export interface Contributor {
  name: string
  email: string
  photoURL: string
  rank: number
}

export interface CreateSecretRequest {
  bundleId: UUID
  name: string
  value: string
}

export interface BundleSecretResponse {
  id?: UUID
  createdAt: Time
  bundleId: UUID
  name: string
}

export interface SecretListResponse {
  page: PageMeta
  data: BundleSecretResponse[]
}

export interface FunctionDefinition {
  image: string
  command: string
  environment?: {[key: string]: string}
  secrets?: string[]
  build?: FunctionBuild
}

export interface JobDefinition {
  notebookPath: string
  environment: {[key: string]: string}
  secrets: string[]
  schedule: string
  description: string
}

export interface FunctionBuild {
  dockerfile?: string
  args?: string[]
  labels?: string[]
  skip?: boolean
}

export interface EditConfig {
  image: string
  environment?: {[key: string]: string}
  secret?: string[]
}

export interface Bundle {
  apiVersion: string
  name: string
  edit: EditConfig
  jobs?: {[key: string]: JobDefinition}
  functions?: {[key: string]: FunctionDefinition}
}

"
`;
